map_test.go

```diff
@@ -2316,37 +2327,101 @@ func ExampleMap_Iterate() {
 		entries = hash.Iterate()
 	)
 
+	m := make(map[string]uint32)
 	for entries.Next(&key, &value) {
 		// Order of keys is non-deterministic due to randomized map seed
-		fmt.Printf("key: %s, value: %d\n", key, value)
+		m[key] = value
 	}
 
 	if err := entries.Err(); err != nil {
 		panic(fmt.Sprint("Iterator encountered an error:", err))
 	}
+
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		fmt.Printf("key: %s, value: %d\n", k, m[k])
+	}
+	// Output: key: hello, value: 21
+	// key: world, value: 42
 }
 
 // It is possible to iterate nested maps and program arrays by
 // unmarshaling into a *Map or *Program.
 func ExampleMap_Iterate_nestedMapsAndProgramArrays() {
-	var arrayOfMaps *Map // Set this up somehow
-
+	spec := &MapSpec{
+		Type: ArrayOfMaps,
+		InnerMap: &MapSpec{
+			Type:       Array,
+			KeySize:    4,
+			ValueSize:  4,
+			MaxEntries: 2,
+		},
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 10,
+	}
+	arrayOfMaps, err := NewMap(spec)
+	if err != nil {
+		if errors.Is(err, internal.ErrNotSupported) {
```

Comment:
Calls to `errors.Is()` don't need to be gated by `err != nil`. Move this block up a line in front of `err != nil`.

Reply:
Why do we need to handle this case? This would need an ancient kernel to fail with ErrNotSupported. Remember, these are supposed to be minimal examples.

Reply:
Fails on CI otherwise due to old kernel testing.

Reply:
Ah, yes, probably the reason why we never committed to making these testable. :)